### PR TITLE
Add Data Access Control

### PR DESCRIPTION
**Pull request by urseberry**
### Merge instructions

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/account_management/rbac/data_access.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+You may create a maximum of 100 restricted datasets. If you need a higher limit, reach out to Support.
+
+### API
+The Data Access Control API is under development and should be considered unstable. Future versions may be backward incompatible. 
+
+Terraform support will be announced after Data Access Control is generally available.
+
+## Select tags for access
+
+Each restricted dataset can control access to multiple types of data, such as metrics. You are free to use the same or different tags across multiple types of telemetry. Within each telemetry type, you must use a _single_ tag or attribute to define your access strategy.
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:68]:
```suggestion
Each Restricted Dataset can control access to multiple types of data, such as metrics. You are free to use the same or different tags across multiple types of telemetry. Within each telemetry type, you must use a _single_ tag or attribute to define your access strategy.
```

File: content/en/account_management/rbac/data_access.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+If you have too many combinations of tags or attributes to fit within these constraints, consider [revisiting your tagging][4] to define a new tag that better reflects your access strategy.
+
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:71]:
```suggestion
# Supported examples and restrictions for Restricted Datasets
```
I think the flow and formatting can be improved here but feel free to take or reject the suggestion

File: content/en/account_management/rbac/data_access.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+**Supported example**
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:72]:
```suggestion
## Supported example
```

File: content/en/account_management/rbac/data_access.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+Restricted dataset 1:
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:74]:
```suggestion
### Restricted dataset 1
```

File: content/en/account_management/rbac/data_access.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+## Select tags for access
+
+Each restricted dataset can control access to multiple types of data, such as metrics. You are free to use the same or different tags across multiple types of telemetry. Within each telemetry type, you must use a _single_ tag or attribute to define your access strategy.
+
+If you have too many combinations of tags or attributes to fit within these constraints, consider [revisiting your tagging][4] to define a new tag that better reflects your access strategy.
+
+**Supported example**
+
+Restricted dataset 1: 
+* Telemetry type: RUM
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:75]:
```suggestion
- **Telemetry Type**: RUM  
   - **Filters**: `@application.id:ABCD`
```

File: content/en/account_management/rbac/data_access.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+
+Each restricted dataset can control access to multiple types of data, such as metrics. You are free to use the same or different tags across multiple types of telemetry. Within each telemetry type, you must use a _single_ tag or attribute to define your access strategy.
+
+If you have too many combinations of tags or attributes to fit within these constraints, consider [revisiting your tagging][4] to define a new tag that better reflects your access strategy.
+
+**Supported example**
+
+Restricted dataset 1: 
+* Telemetry type: RUM 
+    * Filters: `@application.id:ABCD`
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:76]:
```suggestion
```

File: content/en/account_management/rbac/data_access.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+
+Restricted dataset 2:
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:78]:
```suggestion
### Restricted dataset 2
```

File: content/en/account_management/rbac/data_access.md [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+* Telemetry type: RUM 
+    * Filters: `@application.id:ABCD`
+
+Restricted dataset 2: 
+* Telemetry type: RUM
+    * Filters: `@application.id:EFGH`
+* Telemetry type: Metrics
+    * Filters: `env:prod`
+
+**Not supported**
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:84]:
```suggestion
## Not supported examples
```

File: content/en/account_management/rbac/data_access.md [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+
+Restricted dataset 1: 
+* Telemetry type: RUM 
+    * Filters: `@application.id:ABCD`
+
+Restricted Dataset 2: 
+* Telemetry type: RUM
+    * Filters: `@application.id:IJKL` `env:prod`
+
+While this example uses the right `@application.id` tag for RUM as was done for restricted dataset 1, the limit is 1 tag per telemetry type. Instead, consider either creating a restricted dataset with _either_ `application.id` or `env`, or identifying a different tag that better combines these attributes.
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:106]:
```suggestion
This example correctly uses the `@application.id` tag for RUM, as was done for restricted dataset 1. However, the limit is one tag per telemetry type. Instead, consider creating a restricted dataset with _either_ `application.id` or `env`, or identify a different tag that better combines these attributes.
```

File: content/en/account_management/rbac/data_access.md [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+
+Restricted dataset 2: 
+* Telemetry type: RUM
+    * Filters: `@application.id:EFGH`
+* Telemetry type: Metrics
+    * Filters: `env:prod`
+
+**Not supported**
+
+Restricted dataset 1:
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:86]:
```suggestion
### Restricted dataset 1: 
```

File: content/en/account_management/rbac/data_access.md [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+* Telemetry type: Metrics
+    * Filters: `env:prod`
+
+**Not supported**
+
+Restricted dataset 1: 
+* Telemetry type: RUM 
+    * Filters: `@application.id:ABCD`
+
+Restricted dataset 2:
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:90]:
```suggestion
### Restricted dataset 2:
```

File: content/en/account_management/rbac/data_access.md [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+* Telemetry type: RUM 
+    * Filters: `env:prod`
+
+Restricted dataset 1 uses `@application.id` as the tag for RUM data, so a new restricted dataset can't change to a different tag. Instead, consider reconfiguring restricted dataset 2 to use `@application.id`, or changing all of your restricted datasets with RUM data to use another tag.
+
+**Not supported**
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:96]:
```suggestion
## Not supported examples
```

File: content/en/account_management/rbac/data_access.md [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+
+Restricted dataset 1:
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:98]:
```suggestion
### Restricted dataset 1: 
```

File: content/en/account_management/rbac/data_access.md [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+
+Restricted dataset 1 uses `@application.id` as the tag for RUM data, so a new restricted dataset can't change to a different tag. Instead, consider reconfiguring restricted dataset 2 to use `@application.id`, or changing all of your restricted datasets with RUM data to use another tag.
+
+**Not supported**
+
+Restricted dataset 1: 
+* Telemetry type: RUM 
+    * Filters: `@application.id:ABCD`
+
+Restricted Dataset 2:
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:102]:
```suggestion
### Restricted Dataset 2: 
```

File: content/en/account_management/rbac/data_access.md [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+
+### Teams and roles
+
+Data Access Control supports granting access to users through Datadog roles or teams. When granting access, consider your existing access control configuration and access strategy. If you are pursuing a service-based approach and are already [customizing the Service Catalog][9], take advantage of the service ownership model by using Teams as part of your Data Access Control configuration.
+
+**Note:** Teams used for Data Access Control must be configured such that adding or removing users can only be done by team members or an administrator, not `Anyone in the organization`.
+
+## Access enforcement
+
+Users in a Datadog organization with Data Access Control enabled can only see query results for data to which they have access, such as in a Dashboard, in an Explorer, or through the API. A restricted dataset removes access from the data bound in the restricted dataset from users who are not permitted across all Datadog experiences and entry points.
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:157]:
```suggestion
Users in a Datadog organization with Data Access Control enabled can only see query results for data to which they have access, such as in a Dashboard, in an Explorer, or through the API. A restricted dataset removes access to the data defined in the restricted dataset for users who are not permitted, across all Datadog experiences and entry points.
```

File: content/en/account_management/rbac/data_access.md [hunk 16]
```diff
@@ -0,0 +1,183 @@
...
+
+### Data explorers
+
+When exploring Datadog with restrictions enabled, users without permissions are still able to cycle through the list of asset names (for example: applications or metrics), but they do not see query results, top tags, or facet details that have been restricted by restricted datasets. When querying a metric, for example, that contains data which is restricted from the user, a blank graph returns. It appears to the user as if their query does not match any data.
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:161]:
```suggestion
When exploring Datadog with restrictions enabled, users without permissions can still browse the list of asset names (applications or metrics), but they cannot see query results, top tags, or facet details restricted by datasets. For instance, querying a metric with restricted data returns a blank graph, making it appear as if the query does not match any data.
```

Trying to remove the redundancy here with the word restricted in the same sentence :)

File: content/en/account_management/rbac/data_access.md [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+
+### Dashboards and Notebooks
+
+Similar to exploring data in a data explorer like the RUM Explorer or Metrics Explorer, viewing data in dashboards in an organization that has restricted datasets enabled only shows the data the user can access. Since dashboards are shared objects that can be accessed by others, it is possible for two users who have different access to view the same dashboard or notebook at the same time and see different data.
+
+### APIs
+
+When querying data through Datadog APIs with restrictions enabled, users without permissions do not see query results that have been restricted by Restricted Datasets.
```
Comment (aliciascott) [inline content/en/account_management/rbac/data_access.md:169]:
```suggestion
When querying data through Datadog APIs with restrictions enabled, users without permissions do **not** see query results that have been restricted by Restricted Datasets.
```

**Review by aliciascott — CHANGES_REQUESTED** [review]
Hi @urseberry this looks good! I have some suggestions mostly for wording and flow and trying to be consistent with capitalizing or not capitalizing Restricted Dataset. Also I think the Supported and not supported examples could benefit from a little bit of re-formatting, but feel free to keep it as is!

**Comment by urseberry** [comment]
I'm doing the changes in batches to make them easier to reason about. Will come back later to address the rest of the feedback.

**Comment by urseberry** [comment]
Alicia, this is ready for your re-review. Some things to note:
- I accepted all your wording suggestions.
- I decided to make Restricted Dataset capitalized throughout. Thanks for pointing out the inconsistency!
- The in-app link to Data Access Control requires the `org_management` permission, which I don't have in any org, so I can't test it. I assume the PM gave me the right link. Do you have `org_management` anywhere?
- I adjusted your formatting for the examples a bit. I also removed the bolding from the field names, because I thought it made them hard to read. See attached screenshot. Thanks for the suggestion, though. I think the new version is clearer than what I originally had.
Screenshot 2024-11-27 at 12 45 25 AM

File: content/en/account_management/rbac/data_access.md [hunk 18]
```diff
@@ -0,0 +1,183 @@
...
+## Configure data access
+
+Data Access Control lets you create a restricted dataset, which defines a set of data which only users in specified teams or roles can access.
+
+### Datadog site
+
+Log in as a user assigned the Datadog Admin role, or any user with a role in your organization with the [`user_access_manage` permission][5].
+
+1. Navigate to [Organization Settings][6].
+1. On the left side of the page, select [Data Access Controls][7].
```
[the remark this answers is not in the document]
Comment (aliciascott) [reply content/en/account_management/rbac/data_access.md:39]: Ah good point, I do not have that permission either

**Review by aliciascott — APPROVED** [review]
Looks good! Thanks for fixing those things, much easier to read and good flow :)

**Pull request merged**